### ISSUE TITLE
ReusableTasks dependency not resolved automatically for nuget package monotorrent 3.0.0

### ISSUE THREAD

**Issue by jpmikkers** [alanmcgovern/monotorrent#666]
I upgraded the nuget package from some 3.x.x beta to 3.0.0 and then visual studio doesn't understand it also needs to pull in the dependency ReusableTasks 4.0.0. Weirdly, you only notice this at runtime, throws an exception:

    System.TypeInitializationException: 'The type initializer for 'MonoTorrent.Client.ClientEngine' threw an exception.'
    Inner Exception: FileNotFoundException: Could not load file or assembly 'ReusableTasks, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null'. The system cannot find the file specified.

Note: this could very well be a bug in visual studio itself .

Workaround:
Manually add ReusableTasks 4.0.0 nuget package.

**Comment by alanmcgovern** [comment]
That should do the trick!

```
    <dependencies>
      <group targetFramework=".NETFramework4.7.2">
        <dependency id="Mono.Nat" version="3.0.0" exclude="Compile,Build,Analyzers" />
        <dependency id="ReusableTasks" version="4.0.0" exclude="Build,Analyzers" />
        <dependency id="StandardSocketsHttpHandler" version="2.2.0.4" exclude="Build,Analyzers" />
        <dependency id="System.Memory" version="4.5.1" exclude="Build,Analyzers" />
      </group>
      <group targetFramework=".NETCoreApp3.0">
        <dependency id="Mono.Nat" version="3.0.0" exclude="Compile,Build,Analyzers" />
        <dependency id="ReusableTasks" version="4.0.0" exclude="Build,Analyzers" />
        <dependency id="StandardSocketsHttpHandler" version="2.2.0.4" exclude="Build,Analyzers" />
      </group>
      <group targetFramework="net5.0">
        <dependency id="Mono.Nat" version="3.0.0" exclude="Compile,Build,Analyzers" />
        <dependency id="ReusableTasks" version="4.0.0" exclude="Build,Analyzers" />
      </group>
      <group targetFramework="net6.0">
        <dependency id="Mono.Nat" version="3.0.0" exclude="Compile,Build,Analyzers" />
        <dependency id="ReusableTasks" version="4.0.0" exclude="Build,Analyzers" />
      </group>
      <group targetFramework=".NETStandard2.0">
        <dependency id="Mono.Nat" version="3.0.0" exclude="Compile,Build,Analyzers" />
        <dependency id="ReusableTasks" version="4.0.0" exclude="Build,Analyzers" />
        <dependency id="StandardSocketsHttpHandler" version="2.2.0.4" exclude="Build,Analyzers" />
        <dependency id="System.Memory" version="4.5.1" exclude="Build,Analyzers" />
      </group>
      <group targetFramework=".NETStandard2.1">
        <dependency id="Mono.Nat" version="3.0.0" exclude="Compile,Build,Analyzers" />
        <dependency id="ReusableTasks" version="4.0.0" exclude="Build,Analyzers" />
        <dependency id="StandardSocketsHttpHandler" version="2.2.0.4" exclude="Build,Analyzers" />
      </group>
    </dependencies>
    ```
    
    For now manually referencing the additional nuget package is a perfect workaround!

**Comment by alanmcgovern** [comment]
Update shipped and I marked 3.0.0 as deprecated in NuGet! Thanks!
